### PR TITLE
Next auth

### PR DESCRIPTION
**Pull request by SamTolmay**
Closes #ISSUE_NUMBER

## Checklist

- [x] Pull request is made to the "develop" branch
- [ ] Tests added
- [ ] Documentation added/updated
- [x] Code has been formatted with Prettier
- [x] Edits from maintainers are allowed

### CODE REVIEW

File: packages/api/src/auth/getNextAuthConfig.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    throw new Error(operatorErrors[0]);
+  }
+
+  nextAuthConfig.providers = getProviders({ authConfig, plugins });
+
+  // We can either configure this using auth config,
+  // then the user configures this using the _secret operator
+  // -> authConfig.secret = evaluatedAuthConfig.secret;
+  // or we can create a fixed env var/secret name that we read.
+  // -> authConfig.secret = secrets.LOWDEFY_AUTH_SECRET;
```
Comment (Gervwyk) [inline packages/api/src/auth/getNextAuthConfig.js:80]: I prefer the fixed env option, it is easier to explain / document. However, the _secret option is more standard in terms of how we do things for connections.

File: packages/api/src/auth/getNextAuthConfig.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  nextAuthConfig.secret = 'TODO: Configure secret';
+
+  nextAuthConfig.callbacks = callbacks;
+  nextAuthConfig.theme = authConfig.theme;
+  initialized = true;
+  console.log(JSON.stringify(nextAuthConfig, null, 2));
```
Comment (Gervwyk) [inline packages/api/src/auth/getNextAuthConfig.js:86]: Reminder to remove logs.

File: packages/build/src/build/buildAuth/validateAuthConfig.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  const { valid } = validate({
+    schema: lowdefySchema.definitions.authConfig,
+    data: components.auth,
+    returnErrors: true,
+  });
+
+  if (!valid) {
+    throw new Error('lowdefy.auth does not match schema.'); // TODO: Better error message
```
Comment (Gervwyk) [inline packages/build/src/build/buildAuth/validateAuthConfig.js:47]: validate returns the error messages defined in the schema.

Comment (Gervwyk) [reply packages/build/src/build/buildAuth/validateAuthConfig.js:47]: Aren't we doing this test in testSchema?

Comment (SamTolmay) [reply packages/build/src/build/buildAuth/validateAuthConfig.js:47]: We have already logged all the schema errors as warnings, so not sure if we should log them again.

File: packages/server-dev/lib/Page.js [hunk 4]
```diff
@@ -16,23 +16,32 @@
...
 
 const Page = ({ Components, config, pageId, router, types }) => {
+  const { data: session, status } = useSession();
+
+  if (status === 'loading') {
+    return '';
+  }
   const { data: pageConfig } = usePageConfig(pageId, router.basePath);
   if (!pageConfig) {
     router.replace(`/404`);
```
Comment (Gervwyk) [inline packages/server-dev/lib/Page.js:32]: So failed auth will not get pageConfig resulting in redirect to 404?

File: packages/server-dev/pages/api/auth/[...nextauth].js [hunk 5]
```diff
@@ -15,14 +15,17 @@
...
-      issuer: process.env.AUTH0_ISSUER,
-    }),
-  ],
-});
+import { getNextAuthConfig } from '@lowdefy/api';
+
+import authJson from '../../../build/auth.json';
+import providers from '../../../build/plugins/auth/providers.js';
+
+// If getNextAuthConfig needs to be async:
```
Comment (Gervwyk) [inline packages/server-dev/pages/api/auth/[...nextauth].js:23]: can we remove this?

File: packages/server/pages/_app.js [hunk 6]
```diff
@@ -16,12 +16,18 @@
...
 import React from 'react';
 import dynamic from 'next/dynamic';
+import { SessionProvider } from 'next-auth/react';
 
 // Must be in _app due to next specifications.
 import '../build/plugins/styles.less';
 
-function App({ Component, pageProps }) {
-  return <Component {...pageProps} />;
+// TODO: SessionProvider requires basebath
```
Comment (Gervwyk) [inline packages/server/pages/_app.js:24]: import config?

**Review by Gervwyk — APPROVED** [review]

**Review by Gervwyk — APPROVED** [review]

**Pull request merged**